frontend/src/metabase/writeback/containers/CreateDataAppModal/CreateDataAppModal.tsx

```diff
@@ -1,44 +1,83 @@
-import React, { useCallback } from "react";
-import type { LocationDescriptor } from "history";
+import React, { useCallback, useState } from "react";
+import { t } from "ttag";
 import { connect } from "react-redux";
 import { push } from "react-router-redux";
+import type { LocationDescriptor } from "history";
+
+import Button from "metabase/core/components/Button";
 
 import * as Urls from "metabase/lib/urls";
 
-import DataApps from "metabase/entities/data-apps";
+import DataApps, { ScaffoldAppParams } from "metabase/entities/data-apps";
+import { DatabaseSchemaAndTableDataSelector } from "metabase/query_builder/components/DataSelector";
 
-import type { DataApp as IDataApp } from "metabase-types/api";
-import type { State } from "metabase-types/store";
+import type { DataApp } from "metabase-types/api";
+import type { Dispatch, State } from "metabase-types/store";
+
+import {
+  ModalRoot,
+  ModalHeader,
+  ModalTitle,
+  ModalBody,
+  ModalFooter,
+} from "./CreateDataAppModal.styled";
 
 interface OwnProps {
   onClose: () => void;
 }
 
 interface DispatchProps {
+  onCreate: (params: ScaffoldAppParams) => Promise<DataApp>;
   onChangeLocation: (location: LocationDescriptor) => void;
 }
 
 type Props = OwnProps & DispatchProps;
 
-const mapDispatchToProps = {
-  onChangeLocation: push,
-};
-
-function CreateDataAppModal({ onClose, onChangeLocation }: Props) {
-  const handleSave = useCallback(
-    (dataApp: IDataApp) => {
-      onClose();
-      onChangeLocation(Urls.dataApp(dataApp));
+function mapDispatchToProps(dispatch: Dispatch) {
+  return {
+    onCreate: async (params: ScaffoldAppParams) => {
+      const action = await dispatch(DataApps.objectActions.scaffold(params));
+      return DataApps.HACK_getObjectFromAction(action);
     },
-    [onClose, onChangeLocation],
-  );
+    onChangeLocation: (location: LocationDescriptor) =>
+      dispatch(push(location)),
+  };
+}
+
+function CreateDataAppModal({ onCreate, onChangeLocation, onClose }: Props) {
+  const [tableId, setTableId] = useState<number | null>(null);
+
+  const handleCreate = useCallback(async () => {
+    const dataApp = await onCreate({
+      name: t`New App`,
```

Comment:
Makes sense to me. Agnes and @brunobergher tend to avoid it giving it should be easy to rename it after it's created. Check out [this thread](https://metaboat.slack.com/archives/C03SX8EDNC9/p1662547065834229).